.kokoro/deps-finder.sh

```diff
@@ -0,0 +1,202 @@
+  sharedDepsLatestVersion=${sharedDepsLatestVersion/*:}
+  if [[ -z $sharedDepsLatestVersion ]]
+  then
+    echo "ERROR: Failed to find latest shared config version!"
+    return 1
+  fi
+
+  echo "$sharedDepsLatestVersion"
+  return 0
+}
+
+#Pass in dependency to script.
+#Dependency should be of the form artifactId:version
+
+function checkDependency() {
+  dependency=$1
+
+  #If no dependency was passed straight to the script, check our most recent commit for a dep version update.
+  if [[ -z $dependency ]]
+  then
+    dependency=$(getDependency)
+  else
+    if [[ -z $(echo $dependency | grep google-cloud-) ]]
+    then
+      echo "$dependency does not require shared-dependencies criteria. Returning success."
+      return 0
+    fi
+  fi
+
+  #We did not successfully find a valid commit with a dependency. Return success - nothing to check here!
+  if ! [[ $? -eq 0 ]]
+  then
+    echo "Commit message does not have correct formatting. Assuming no error."
+    return 0
+  fi
+
+  #Should just be the desired version for shared-deps, e.g. "0.8.3"
+  desiredDeps=$2
+
+  if [[ -z $desiredDeps ]]
+  then
+    desiredDeps=$(getLatestSharedDepsVersion)
+    if [[ -z $desiredDeps ]]
+    then
+      echo "Unable to find correct version of shared-dependencies."
+      return 1
+    fi
+  fi
+
+  depVersion=${dependency/*:} #Grab last part of dependency artifactId:version input.
+  depVersion=${depVersion/-*} #Remove any alphabetical characters at the end (e.g. -alpha, -beta, etc)
+
+  pomLocation="/pom.xml"
+
+  #The naming of these libraries does not follow the rules. As such, we've added in special cases for them.
+  #depAID=dependency artifact ID - this is the ID inside of the Github link for the dependency.
+  case $dependency in
+    *"google-cloud-bigtable-bom"*)
+    depAID="java-bigtable"
+    pomLocation="/google-cloud-bigtable-deps-bom/pom.xml"
+    ;;
+    *"google-cloud-core-bom"*)
+    #This is an exceptional case which does not require shared-dependencies! Return true automatically.
+    echo "Found exception google-cloud-core-bom!"
+    return 0
+    ;;
+    *"google-cloud-build-bom"*)
+    depAID="java-cloudbuild"
+    ;;
+    *"google-cloud-monitoring-dashboard-bom"*)
+    depAID="java-monitoring-dashboards"
+    ;;
+    *"google-cloud-nio"*)
+    depAID="java-storage-nio"
+    ;;
+    *)
+    #General rules for obtaining the Github repo name are:
+    #Replace google-cloud with java, (of course) remove the version number, and remove -bom from the end (if there)
+    depAID=${dependency/*google-cloud-/java-}
+    depAID=${depAID/:*}
+    depAID=${depAID/-bom*}
+    ;;
+  esac
+
+  #Grab raw POM file from Github
+  rawFileStart="https://raw.githubusercontent.com/googleapis/"
+  gitPom=$rawFileStart$depAID"/v"$depVersion$pomLocation
+
+  #If $3 is nonempty, we're running this script directly.
+  if [[ ! -z $3 ]]
+  then
+    echo "POM File Link Generated: $gitPom"
+  fi
+
+  content=$(wget $gitPom -q -O -)
```

Comment:
nit: curl is more commonly available by default on unix distributions (e.g. ubuntu usually doesn't include wget in its default install).

Reply:
Testing with curl, I unfortunately ran into stability issues. When using curl, the script may fail to read some POMs from online (typically 1 per run of the script), and takes about 4-5x the time to run when compared with wget